File: Patterns/C#/Structural/Composite/Boss.cs
using System;
using System.Collections.Generic;

public class Boss : Worker{

    private List<Worker> workers;

    public Boss(string name): base(name) {
        workers = new List<Worker>();
    }

    public override void add(Worker worker){
        workers.Add(worker);
    }

    public override void remove(Worker worker){
        workers.Remove(worker);
    }

    public override void displayStructure(){
        Console.WriteLine(name);

        foreach(Worker worker in workers)
            worker.displayStructure();
    }

}

File: Patterns/Java/Structural/Composite/Boss.java
import java.util.List;
import java.util.ArrayList;

public class Boss extends Worker{

    private List<Worker> workers;

    public Boss(String name){
        super(name);
        workers = new ArrayList<Worker>();
    }

    public void add(Worker worker){
        workers.add(worker);
    }

    public void remove(Worker worker){
        workers.remove(worker);
    }

    public void displayStructure(){
        System.out.println(name);

        for(Worker worker : workers)
            worker.displayStructure();
    }

}
